frontend/dockerfile/linter/linter.go

```diff
@@ -5,16 +5,82 @@ import (
 	"strings"
 
 	"github.com/moby/buildkit/frontend/dockerfile/parser"
+	"github.com/pkg/errors"
 )
 
+type RuleChecker struct {
+	SkippedRules       map[string]struct{}
+	CalledRules        map[string]struct{}
+	SkipAll            bool
+	ErrorFromViolation bool
+	FromBuildArg       bool
+	Warn               LintWarnFunc
+}
+
+func NewRuleChecker() *RuleChecker {
+	return &RuleChecker{
+		SkippedRules: map[string]struct{}{},
+		CalledRules:  map[string]struct{}{},
+	}
+}
+
+func NewRuleCheckerFromStr(lintOptionStr string) (*RuleChecker, error) {
```

Comment:
Don't initialize with unparsed string. Helper in `dockerfile2llb` to create opt struct from string and then only one `New()` function in here that is initialized with this opt (or zero value of it)